### PR TITLE
[TA] Implementation to support FHIR

### LINKED ISSUE #28328: Enable FHIR parameter into existing Healthcare endpoint
Swagger information => https://github.com/Azure/azure-rest-api-specs/pull/17710

### PR DESCRIPTION
**Pull request by maririos**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/28328

### CODE REVIEW

**Review by deyaaeldeen — APPROVED** [review]
Looks good. Perhaps one nitpicky thing to consider is mention a thing or two about FHIR protocol in the changelog to educate customers about the feature.

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs [hunk 1]
```diff
@@ -443,6 +444,34 @@ public async Task AnalyzeHealthcareEntitiesPagination()
...
         }
 
+        [RecordedTest]
+        public async Task RecognizeHealthcareEntitiesWithFhir()
+        {
+            TextAnalyticsClient client = GetClient(useTokenCredential: true);
+
+            AnalyzeHealthcareEntitiesOptions options = new AnalyzeHealthcareEntitiesOptions()
+            {
+                FhirVersion = FhirVersion.Four01
```
Comment (kristapratico) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs:454]: I'm guessing this is just how it generates?

**Review by kristapratico — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/ServiceClients/LanguageServiceClient.cs [hunk 2]
```diff
@@ -1286,7 +1286,8 @@ private static HealthcareLROTask CreateHealthcareTask(AnalyzeHealthcareEntitiesO
                 {
                     ModelVersion = options.ModelVersion,
                     StringIndexType = Constants.DefaultStringIndexType,
-                    LoggingOptOut = options.DisableServiceLogs
+                    LoggingOptOut = options.DisableServiceLogs,
```
Comment (jsquire) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/ServiceClients/LanguageServiceClient.cs:1289]: Does this need to be reflected to the Legacy API?

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs (hunk 1 above)
Comment (jsquire) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs:454]: woah.

**Review by jsquire — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs (hunk 1 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs:454]: Yes :( Issue https://github.com/Azure/azure-sdk-for-net/issues/28811 to make it better next release :)

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/ServiceClients/LanguageServiceClient.cs (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/ServiceClients/LanguageServiceClient.cs:1289]: No. it is an addition that was introduced only in the Language service.
There is no equivalent on where to assign it if the user passes a previous service version. If they still assign the value and target a previous version, the SDK drops the property

Comment (jsquire) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/ServiceClients/LanguageServiceClient.cs:1289]: Is this the first case of us hitting the "how do we handle mismatch errors?"    Does the behavior change or trigger anything else strange for the legacy surface?    Should we drop note in #28060 to mention this scenario for follow-up?

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs [hunk 3]
```diff
@@ -443,6 +444,35 @@ public async Task AnalyzeHealthcareEntitiesPagination()
             Assert.AreEqual(2, pages[0].Count);
         }
 
+        [RecordedTest]
+        [ServiceVersion(Min = TextAnalyticsClientOptions.ServiceVersion.V2022_04_01_Preview)]
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs:448]: Fixed to only run for new versions

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs [hunk 4]
```diff
@@ -497,6 +527,8 @@ private void ValidateBatchDocumentsResult(AnalyzeHealthcareEntitiesResultCollect
                 }
 
                 Assert.IsNotNull(entitiesInDocument.Warnings);
+                Assert.IsNotNull(entitiesInDocument.EntityRelations);
+                Assert.IsNotNull(entitiesInDocument.FhirBundle);
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs:531]: This validation happens in most of the tests (version independent)

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/ServiceClients/LanguageServiceClient.cs (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/ServiceClients/LanguageServiceClient.cs:1289]: > Is this the first case of us hitting the "how do we handle mismatch errors?" 

It indeed is :)

>  Does the behavior change or trigger anything else strange for the legacy surface?

It does not. The `FhriVersion` input property only affects the `FhirBundle` output property which was introduced in this new version too.

>Should we drop note in https://github.com/Azure/azure-sdk-for-net/issues/28060 to mention this scenario for follow-up?

Added. I'm following what we have done in this library when a new property is added that is only supported in specific versions and does not affect behavior of older version output properties:
- We add that information to docstrings for both input and output property.
- Input property:
  - If the property is a query parameter, we send it to the service and let them manage it.
  - If the property is a body parameter, we drop it.
- Output property: Value is always empty Dict even when targeting older versions (I added comments in the tests to pint this out)

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs (hunk 4 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs:531]: Running official build to run tests from all versions

**Review by mssfang — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs (hunk 4 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs:531]: yay for tests!. looking into it

Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs:531]: Fixed

**Comment by maririos** [comment]
@jsquire My last commit includes the fix for this to work in both legacy and language clients. To keep it consistent with the work you did in legacy, I decided on not touching `Transforms.Legacy` and instead, having another constructor that takes the new parameters. Apologies for the iterations in the PR while figuring out best approach.
Let me know what you think.

**Review by jsquire — APPROVED** [review]

**Pull request merged**